wallet.md

```diff
@@ -0,0 +1,144 @@
+Wallet API
+==========
+
+BlockAdded Notifications
+------------------------
+```
+type NotifyBlockAddedRequestMessage struct {}
+
+type NotifyBlockAddedResponseMessage struct {
+	Error *RPCError
+}
+
+type BlockAddedNotificationMessage struct {
+	Block *RPCBlock
+}
+```
+ChainChanged Notifications
+--------------------------
+```
+type NotifyChainChangedRequestMessage struct {}
+
+type NotifyChainChangedResponseMessage strcut {
+	Error *RPCError
+}
+
+type ChainChangedNotificationMessage struct {
+	RemovedChainBlockHashes []string
+	AddedChainBlocks []*ChainBlock
+}
+
+type ChainBlock struct {
+	Hash string
+	AcceptedBlocks []*AcceptedBlock
+}
+
+type AcceptedBlock struct {
+	Hash string
+	AcceptedTransactionIDs []string
+}
+```
+UTXOsChanged Notifications
+--------------------------
+```
+type NotifyUTXOsChangedRequestMessage struct {
+	Addresses []string
+}
+
+type NotifyUTXOsChangedResponseMessage struct {
+	Error *RPCError
+}
+
+type UTXOsChangedNotificationMessage struct {
+	Added []*UTXOsByAddressesEntry
+	Removed []*RPCOutpoint
+}
+```
+GetUTXOsByAddresses
+-------------------
+```
+type GetUTXOsByAddressesRequestMessage struct {
+	Addresses []string
+}
+
+type GetUTXOsByAddressesResponseMessage struct {
+	Entries []*UTXOsByAddressesEntry
+
+	Error *RPCError
+}
+
+type UTXOsByAddressesEntry struct {
+	Address string
+	Outpoint *RPCOutpoint
+	UTXOEntry *RPCUTXOEntry
+}
+```
+SubmitTransaction
+-----------------
+```
+type SubmitTransactionRequestMessage struct {
+	Transaction *RPCTransaction
+}
+
+type SubmitTransactionResponseMessage struct {
+    TransactionID string
+    
+    Error *RPCError
+}
+```
+Dependant Types
+---------------
+```
+type RPCTransaction struct {
+	Version int32
+	Inputs []*RPCTransactionInput
+	Outputs []*RPCTransactionOutput
+	LockTime uint64
+	SubnetworkID string
+	Gas uint64
+	PayloadHash string
+	Payload string
+	ID string
+	Hash string
+}
+
+type RPCTransactionInput struct {
+	PreviousOutpoint *RPCOutpoint
+	SignatureScript string
+	Sequence uint64
+}
+
+type RPCTransactionOutput struct {
+	Amount uint64
+	ScriptPubKey string
+}
+
+type RPCOutpoint struct {
+	TransactionID string
+	Index uint32
+}
+
+type RPCUTXOEntry struct {
+	Amount uint64
+	ScriptPubKey string
+	BlockBlueScore uint64
```

Comment:
We probably need IsCoinbase here as well.
Otherwise wallet will try to spend coinbase entries before maturity